src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -0,0 +1,128 @@
+
+    The input can be either a raw waveform or a audio file. In case of the audio file, ffmpeg should
+    be installed for the read function requires it (to support many formats).
+    """
+
+    def __init__(self, feature_extractor: "AutoFeatureExtractor", *args, **kwargs):
+        """
+        Arguments:
+            model (:obj:`~transformers.PreTrainedModel` or :obj:`~transformers.TFPreTrainedModel`):
+                The model that will be used by the pipeline to make predictions. This needs to be a model inheriting from
+                :class:`~transformers.PreTrainedModel` for PyTorch and :class:`~transformers.TFPreTrainedModel` for
+                TensorFlow.
+            tokenizer (:obj:`~transformers.PreTrainedTokenizer`):
+                The tokenizer that will be used by the pipeline to encode data for the model. This object inherits from
+                :class:`~transformers.PreTrainedTokenizer`.
+            feature_extractor (:obj:`~transformers.AutoFeatureExtractor`):
+                The feature extractor that will be used by the pipeline to encode waveform for the model.
+            modelcard (:obj:`str` or :class:`~transformers.ModelCard`, `optional`):
+                Model card attributed to the model for this pipeline.
+            framework (:obj:`str`, `optional`):
+                The framework to use, either :obj:`"pt"` for PyTorch or :obj:`"tf"` for TensorFlow. The specified framework
+                must be installed.
+
+                If no framework is specified, will default to the one currently installed. If no framework is specified and
+                both frameworks are installed, will default to the framework of the :obj:`model`, or to PyTorch if no model
+                is provided.
+            device (:obj:`int`, `optional`, defaults to -1):
+                Device ordinal for CPU/GPU supports. Setting this to -1 will leverage CPU, a positive will run the model on
+                the associated CUDA device id.
+        """
+        super().__init__(*args, **kwargs)
+        self.feature_extractor = feature_extractor
+
+    def __call__(
+        self,
+        inputs: Union[np.ndarray, bytes, str],
+        **kwargs,
+    ):
+        """
+        Classify the sequence(s) given as inputs. See the :obj:`~transformers.AutomaticSpeechRecognitionPipeline`
+        documentation for more information.
+
+        Args:
+            inputs (:obj:`np.ndarray` or :obj:`bytes` or :obj:`str`):
+                The inputs is either a raw waveform (:obj:`np.ndarray`) at the correct sampling rate (no further check will be done) or a :obj:`str` that is
+                the filename of the audio file, the file will be read at the correct sampling rate to get the waveform using `ffmpeg`.
+                This requires `ffmpeg` to be installed on the system. If `inputs` is :obj:`bytes` it is supposed to be the content of an audio file and is
+                interpreted by `ffmpeg` in the same way.
```

Comment:
The pipeline doesn't handle batching - could the expected dimensions of this `inputs` variable when passed in as an `np.ndarray` be explicit?

Reply:
Well I think pipelines should try to avoid doing real batching anyway : 
https://github.com/huggingface/transformers/pull/11251#pullrequestreview-637488364

I could add the `list` behavior that exists ìn other pipelines, but I wouldn't implement it as batching (here padding will be much worse than with text).

I really full funny implementing :
```python
if not isinstance(xX, list): 
    inputs = [inputs]
.... 
for input_ in inputs: 
    results.append(do_something(input_))

if len(results) == 1:
   results = results[0]
```
everywhere. It's error-prone and confusing to me. For instance it will return an object and not a list if I passe a single item list as input, where I would expect to receive a single list output.

Because this pipeline does not currently exist, I was hesitant to put this directly here. There is no backward compatibility yet, I wanted to avoid introducing something which feels like a bad pattern.
Coherence is also important, so I can definitely add it if coherence trumps correctness in your opinion.

Reply:
I definitely understand your comment regarding batching, I wasn't proposing enabling batching on the pipeline - just specifying what are the expected dimensions of `inputs`!

Reply:
You mean adding so asserts on dimensions ?

Reply:
Just the docs I guess :)